runtime/src/deterministic.rs

```diff
@@ -214,7 +253,14 @@ impl Executor {
     /// The cycle duration determines how much time is advanced after each iteration of the event
     /// loop. This is useful to prevent starvation if some task never yields.
     pub fn init(seed: u64, cycle: Duration) -> (Runner, Context, Arc<Auditor>) {
+        let metrics = Arc::new(Metrics {
+            bandwidth_usage: Family::default(),
+            tasks_spawned: Counter::default(),
+            task_polls: Counter::default(),
+        });
+
         let auditor = Arc::new(Auditor::new());
+        let registry = Arc::new(Mutex::new(Registry::default()));
```

Comment:
I have an issue to migrate `init` to use a `Config` (https://github.com/commonwarexyz/monorepo/issues/88) and will clean this up there...but could you modify the code to accept `registry: Arc<Mutex<Registry>>` as a parameter?

Reply:
currently on it, but this will need refactoring across places that use the init func, for the tests I can just pass the `Arc::new(Mutex::new(Registry::default()))` should I do that for the rest

Reply:
Yeah, you can do that everywhere 👍 .

Reply:
I'd prefer you use: `Arc::new(Mutex::new(Registry::with_prefix("runtime")));`

(like in https://github.com/commonwarexyz/monorepo/blob/9109b98702651b5521104251f98871160efcf396/examples/chat/src/main.rs#L143)